addons/mail/static/tests/helpers/mock_server/models/mail_activity.js

```diff
@@ -118,57 +126,167 @@ patch(MockServer.prototype, {
-        });
-        groups.forEach(function (group) {
-            // mockReadGroup doesn't correctly return all asked fields
-            const activites = self.getRecords("mail.activity", group.__domain);
-            group.activity_type_id = group.activity_type_id[0];
-            let minDate;
-            activites.forEach(function (activity) {
-                if (!minDate || deserializeDate(activity.date_deadline) < deserializeDate(minDate)) {
-                    minDate = activity.date_deadline;
-                }
-            });
-            group.date_deadline = minDate;
-            resIdToDeadline[group.res_id] = minDate;
-            let state;
-            if (group.date_deadline === serializeDate(DateTime.now())) {
-                state = "today";
-            } else if (deserializeDate(group.date_deadline) > DateTime.now()) {
-                state = "planned";
-            } else {
-                state = "overdue";
+        const resIdToDateDone = {};
+        const groupedActivities = {};
+        for (const resIdActivityTypeIdStr of new Set([
+            ...Object.keys(groupedCompletedActivities),
+            ...Object.keys(groupedOngoingActivities),
+        ])) {
+            const [resId, activityTypeId] = resIdActivityTypeIdStr.split(",").map((n) => Number(n));
+            const ongoingActivities = groupedOngoingActivities[resIdActivityTypeIdStr] || [];
+            const completedActivities = groupedCompletedActivities[resIdActivityTypeIdStr] || [];
+            const dateDone = completedActivities.length
+                ? DateTime.max(...completedActivities.map((a) => deserializeDate(a.date_done)))
+                : false;
+            const dateDeadline = ongoingActivities.length
+                ? DateTime.min(...ongoingActivities.map((a) => deserializeDate(a.date_deadline)))
+                : false;
+            if (
+                dateDeadline &&
+                (resIdToDeadline[resId] === undefined || dateDeadline < resIdToDeadline[resId])
+            ) {
+                resIdToDeadline[resId] = dateDeadline;
+            }
+            if (
+                dateDone &&
+                (resIdToDateDone[resId] === undefined || dateDone > resIdToDateDone[resId])
+            ) {
+                resIdToDateDone[resId] = dateDone;
             }
-            if (!groupedActivities[group.res_id]) {
-                groupedActivities[group.res_id] = {};
+            const distinctAssignees = unique(
+                sortBy(
+                    ongoingActivities.filter((a) => a.user_id),
+                    (a) => a.date_deadline
+                ).map((a) => a.user_id)
+            );
+            const closestDate = ongoingActivities.length ? dateDeadline : dateDone;
+            const activityAttachments = completedActivities
+                .map((activity) => activity.attachment_ids)
+                .flat()
+                .map((attachmentId) => attachmentsById[attachmentId]);
```

Comment:
Example of updated naming to gain some lines and ease reading
```suggestion
            const activityAttach = completed.map((act) => act.attachment_ids).flat()
                .map((attachId) => attachmentsById[attachId]);
```